man1/aegis.1
.\"
.\"	aegis - project change supervisor
.\"	Copyright (C) 1991, 1992, 1993 Peter Miller.
.\"	All rights reserved.
.\"
.\"	This program is free software; you can redistribute it and/or modify
.\"	it under the terms of the GNU General Public License as published by
.\"	the Free Software Foundation; either version 2 of the License, or
.\"	(at your option) any later version.
.\"
.\"	This program is distributed in the hope that it will be useful,
.\"	but WITHOUT ANY WARRANTY; without even the implied warranty of
.\"	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
.\"	GNU General Public License for more details.
.\"
.\"	You should have received a copy of the GNU General Public License
.\"	along with this program; if not, write to the Free Software
.\"	Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
.\"
.\" MANIFEST: manual entry for aegis
.\"
.so z_name.so
.TH \*(n) 1 \*(N)
.SH NAME
\*(n) \- project change supervisor
.SH SYNOPSIS
.B \*(n)
.I function
[
.IR option ...
]
.br
.B \*(n)
.B -Help
.SH DESCRIPTION
The
.I \*(n)
program is used to
supervise the development and integration of changes into projects.
.SH FUNCTIONS
The following functions are available:
.TP 8n
.B -Build
.br
The 
.I "\*(n) -Build"
command is used to
build a project.
See
.IR aeb (1)
for more information.
.TP 8n
.B -Change_Attributes
.br
The 
.I "\*(n) -Change_Attributes"
command is used to
modify the attributes of a change.
See
.IR aeca (1)
for more information.
.TP 8n
.B -Change_Directory
.br
The 
.I "\*(n) -Change_Directory"
command is used to
change directory.
See
.IR aecd (1)
for more information.
.TP 8n
.B -CoPy_file
.br
The 
.I "\*(n) -CoPy_file"
command is used to
copy a file into a change.
See
.IR aecp (1)
for more information.
.TP 8n
.B -CoPy_file_Undo
.br
The 
.I "\*(n) -Copy_File_Undo"
command is used to
remove a copy of a file from a change.
See
.IR aecpu (1)
for more information.
.TP 8n
.B -Develop_Begin
.br
The 
.I "\*(n) -Develop_Begin"
command is used to
begin development of a change.
See
.IR aedb (1)
for more information.
.TP 8n
.B -Develop_Begin_Undo
.br
The 
.I "\*(n) -Develop_Begin_Undo"
command is used to
cease development of a change.
.See
.IR aedbu (1)
for more information.
.TP 8n
.B -Develop_End
.br
The 
.I "\*(n) -Develop_End"
command is used to
complete development of a change.
See
.IR aede (1)
for more information.
.TP 8n
.B -Develop_End_Undo
.br
The 
.I "\*(n) -Develop_End_Undo"
command is used to
recall a change for further development.
See
.IR aedeu (1)
for more information.
.TP 8n
.B -DIFFerence
.br
The 
.I "\*(n) -DIFFerence"
command is used to
find differences between development directory and baseline.
See
.IR aed (1)
for more information.
.so o_help.so
.TP 8n
.B -Integrate_Begin
.br
The 
.I "\*(n) -Integrate_Begin"
command is used to
being integrating a change.
See
.IR aeib (1)
for more information.
.TP 8n
.B -Integrate_Begin_Undo
.br
The 
.I "\*(n) -Integrate_Begin_Undo"
command is used to
cease integrating a change.
See
.IR aeibu (1)
for more information.
.TP 8n
.B -Integrate_Fail
.br
The 
.I "\*(n) -Integrate_Fail"
command is used to
fail a change integration.
See
.IR aeif (1)
for more information.
.TP 8n
.B -Integrate_Pass
.br
The 
.I "\*(n) -Integrate_Pass"
command is used to
pass a change integration.
See
.IR aeip (1)
for more information.
.TP 8n
.B -List
.br
The 
.I "\*(n) -List"
command is used to
list interesting things.
See
.IR ael (1)
for more information.
.TP 8n
.B -MoVe_file
.br
The 
.I "\*(n) -MoVe_file"
command is used to
change the name of a file
as part of a change.
See
.IR aemv (1)
for more information.
.TP 8n
.B -New_Administrator
.br
The 
.I "\*(n) -New_Administrator"
command is used to
add new administrators to a project.
See
.IR aena (1)
for more information.
.TP 8n
.B -New_Change
.br
The 
.I "\*(n) -New_Change"
command is used to
add a new change to a project.
See
.IR aenc (1)
for more information.
.TP 8n
.B -New_Change_Undo
.br
The 
.I "\*(n) -New_Change_Undo"
command is used to
remove a new change from a project.
See
.IR aencu (1)
for more information.
.TP 8n
.B -New_Developer
.br
The 
.I "\*(n) -New_Developer"
command is used to add
new developers to a project.
See
.IR aend (1)
for more information.
.TP 8n
.B -New_File
.br
The 
.I "\*(n) -New_File"
command is used to
add new files to a change.
See
.IR aenf (1)
for more information.
.TP 8n
.B -New_File_Undo
.br
The 
.I "\*(n) -New_File_Undo"
command is used to
remove new files from a change.
See
.IR aenfu (1)
for more information.
.TP 8n
.B -New_Integrator
.br
The 
.I "\*(n) -New_Integrator"
command is used to
add new integrators to a project.
See
.IR aeni (1)
for more information.
.TP 8n
.B -New_Project
.br
The 
.I "\*(n) -New_Project"
command is used to
create a new project to be watched over by \*(n).
See
.IR aenpr (1)
for more information.
.TP 8n
.B -New_ReLeaSe
.br
The 
.I "\*(n) -New_ReLeaSe"
command is used to
create a new project from an existing project.
See
.IR aenrls (1)
for more information.
.TP 8n
.B -New_ReViewer
.br
The 
.I "\*(n) -New_ReViewer"
command is used to
add new reviewers to a project.
See
.IR aenrv (1)
for more information.
.TP 8n
.B -New_Test
.br
The 
.I "\*(n) -New_Test"
command is used to
add a new test to a change
See
.IR aent (1)
for more information.
.TP 8n
.B -New_Test_Undo
.br
The
.I "\*(n) -New_Test_Undo"
command is used to
remove new tests from a change.
See
.IR aentu (1)
for more information.
.TP 8n
.B -Project_Attributes
.br
The 
.I "\*(n) -Project_Attributes"
command is used to
modify the attributes of a project.
.TP 8n
.B -Remove_Administrator
.br
The 
.I "\*(n) -Remove_Administrator"
command is used to
remove administrators from a project.
See
.IR aera (1)
for more information.
.TP 8n
.B -Remove_Developer
.br
The 
.I "\*(n) -Remove_Developer"
command is used to
remove developers from a project.
See
.IR aerd (1)
for more information.
.TP 8n
.B -ReMove_file
.br
The 
.I "\*(n) -ReMove_file"
command is used to
add files to be deleted to a change.
See
.IR aerm (1)
for more information.
.TP 8n
.B -ReMove_file_Undo
.br
The 
.I "\*(n) -Remove_File_Undo"
command is used to
remove files to be deleted from a change.
See
.IR aermu (1)
for more information.
.TP 8n
.B -Remove_Integrator
.br
The 
.I "\*(n) -Remove_Integrator"
command is used to
remove integrators from a project.
See
.IR aeri (1)
for more information.
.TP 8n
.B -ReMove_PRoject
.br
The 
.I "\*(n) -ReMove_PRoject"
command is used to
remove a project.
See
.IR aermpr (1)
for more information.
.TP 8n
.B -Remove_ReViewer
.br
The 
.I "\*(n) -Remove_ReViewer"
command is used to
remove reviewers from a project.
See
.IR aerrv (1)
for more information.
.TP 8n
.B -Review_Fail
.br
The 
.I "\*(n) -Review_Fail"
command is used to
fail a change review.
See
.IR aerf (1)
for more information.
.TP 8n
.B -Review_Pass
.br
The 
.I "\*(n) -Review_Pass"
command is used to
pass a change review.
See
.IR aerp (1)
for more information.
.TP 8n
.B -Review_Pass_Undo
.br
The 
.I "\*(n) -Review_Pass_Undo"
command is used to
rescind a change review pass.
See
.IR aerpu (1)
for more information.
.TP 8n
.B -Test
.br
The 
.I "\*(n) -Test"
command is used to
run tests.
See
.IR aet (1)
for more information.
.TP 8n
.B -VERSion
.br
The
.I "\*(n) -VERSion"
command is used to
get copyright and version details.
See
.IR aev (1)
for more information.
.PP
All function are case insensitive.
Functions may be abbreviated;
the abbreviation is the upper case letters.
Functions must appear as the first command line argument.
.SH OPTIONS
The following options are available to all functions.
These options may appear anywhere on the command line following
the function selectors.
.so o_lib.so
.so o_page.so
.so o_terse.so
.so o_unforma.so
.so o_verbose.so
.so o__rules.so
.so z_exit.so
.so z_cr.so


etc/CHANGES



Project "aegis"								  Page 1
Details							Sat Apr	 5 23:36:53 2003

NAME
	Project "aegis".

SUMMARY
	project change supervisor.

DESCRIPTION
	project change supervisor.

BRANCH CONTENTS
	Change	State		Description
	------- -------		-----------
	3	completed	project change supervisor, branch 3.
	4	being_developed project change supervisor, branch 4.

ARCHITECTURES
	This change must be developed on the unspecified architectures.

	arch.	 host	  aeb	      aet	  aet -bl     aet -reg
	-------- -------- ---------   ---------	  ---------   ---------
	unspecif	  required    required	  required    exempt
	ied

CAUSE
	This change was caused by internal_enhancement.

FILES
	Type	Action	 Edit	      File Name
	------	------	 ------	      ------
	build	remove	 5	      BUILDING
				      Deleted by change 3.
	source	remove	 2 (5)	      LICENSE
				      Deleted by change 3.
	build	remove	 8 (5)	      MANIFEST
				      Deleted by change 3.
	build	remove	 10 (5)	      Makefile.in
				      Deleted by change 3.
	build	remove	 6 (5)	      README
				      Deleted by change 3.
	source	create	 7 (5)	      aedist/arglex3.c
	source	create	 7 (5)	      aedist/arglex3.h
	source	remove	 4 (5)	      aedist/header.c
				      Deleted by change 3.
				      Moved to libaegis/rfc822header.c
	source	remove	 1 (5)	      aedist/header.h
				      Deleted by change 3.
				      Moved to libaegis/rfc822header.h
	source	remove	 5	      aedist/input/822wrap.c
				      Deleted by change 3.
				      Moved to libaegis/input/822wrap.c
	source	remove	 2 (5)	      aedist/input/822wrap.h
				      Deleted by change 3.
				      Moved to libaegis/input/822wrap.h
	source	create	 3 (5)	      aedist/input/cpio.c
	source	create	 1 (5)	      aedist/input/cpio.h
	source	create	 5	      aedist/input/cpio_child.c
	source	create	 1 (5)	      aedist/input/cpio_child.h



Project "aegis"								  Page 2
Details							Sat Apr	 5 23:36:53 2003

	Type	Action	 Edit	      File Name
	------	------	 ------	      ------
	source	create	 9 (5)	      aedist/list.c
	source	create	 1 (5)	      aedist/list.h
	source	create	 5	      aedist/main.c
	source	create	 7 (5)	      aedist/open.c
	source	create	 2 (5)	      aedist/open.h
	source	create	 4 (5)	      aedist/output/cpio.c
	source	create	 1 (5)	      aedist/output/cpio.h
	source	create	 5	      aedist/output/cpio_child.c
	source	create	 1 (5)	      aedist/output/cpio_child.h
	source	create	 4 (5)	      aedist/output/cpio_child2.c
	source	create	 1 (5)	      aedist/output/cpio_child2.h
	source	create	 29 (5)	      aedist/receive.c
	source	create	 1 (5)	      aedist/receive.h
	source	create	 21 (5)	      aedist/send.c
	source	create	 1 (5)	      aedist/send.h
	source	create	 2 (5)	      aefind/cmdline.h
	source	create	 8 (5)	      aefind/cmdline.y
	source	create	 4 (5)	      aefind/descend.c
	source	create	 2 (5)	      aefind/descend.h
	source	create	 3 (5)	      aefind/function.c
	source	create	 2 (5)	      aefind/function.h
	source	create	 2 (5)	      aefind/function/basename.c
	source	create	 2 (5)	      aefind/function/basename.h
	source	create	 3 (5)	      aefind/function/needs.c
	source	create	 2 (5)	      aefind/function/needs.h
	source	create	 2 (5)	      aefind/function/print.c
	source	create	 2 (5)	      aefind/function/print.h
	source	create	 4 (5)	      aefind/function/stat.c
	source	create	 2 (5)	      aefind/function/stat.h
	source	create	 5	      aefind/lex.c
	source	create	 2 (5)	      aefind/lex.h
	source	create	 2 (5)	      aefind/main.c
	source	create	 2 (5)	      aefind/shorthand/name.c
	source	create	 2 (5)	      aefind/shorthand/name.h
	source	create	 2 (5)	      aefind/shorthand/path.c
	source	create	 2 (5)	      aefind/shorthand/path.h
	source	create	 2 (5)	      aefind/shorthand/print.c
	source	create	 2 (5)	      aefind/shorthand/print.h
	source	create	 3 (5)	      aefind/shorthand/stat.c
	source	create	 2 (5)	      aefind/shorthand/stat.h
	source	create	 2 (5)	      aefind/tree.c
	source	create	 2 (5)	      aefind/tree.h
	source	create	 4 (5)	      aefind/tree/arithmetic.c
	source	create	 2 (5)	      aefind/tree/arithmetic.h
	source	create	 3 (5)	      aefind/tree/bitwise.c
	source	create	 2 (5)	      aefind/tree/bitwise.h
	source	create	 2 (5)	      aefind/tree/constant.c
	source	create	 2 (5)	      aefind/tree/constant.h
	source	create	 2 (5)	      aefind/tree/diadic.c
	source	create	 2 (5)	      aefind/tree/diadic.h
	source	create	 2 (5)	      aefind/tree/list.c
	source	create	 2 (5)	      aefind/tree/list.h
	source	create	 3 (5)	      aefind/tree/logical.c
	source	create	 2 (5)	      aefind/tree/logical.h
	source	create	 3 (5)	      aefind/tree/match.c



Project "aegis"								  Page 3
Details							Sat Apr	 5 23:36:53 2003

	Type	Action	 Edit	      File Name
	------	------	 ------	      ------
	source	create	 2 (5)	      aefind/tree/match.h
	source	create	 2 (5)	      aefind/tree/monadic.c
	source	create	 2 (5)	      aefind/tree/monadic.h
	source	create	 2 (5)	      aefind/tree/now.c
	source	create	 2 (5)	      aefind/tree/now.h
	source	create	 2 (5)	      aefind/tree/private.c
	source	create	 2 (5)	      aefind/tree/private.h
	source	create	 4 (5)	      aefind/tree/relative.c
	source	create	 2 (5)	      aefind/tree/relative.h
	source	create	 2 (5)	      aefind/tree/this.c
	source	create	 2 (5)	      aefind/tree/this.h
	source	create	 3 (5)	      aefp/main.c
	source	create	 17 (5)	      aegis/aeb.c
	source	create	 2 (5)	      aegis/aeb.h
	source	create	 13 (5)	      aegis/aeca.c
	source	create	 3 (5)	      aegis/aeca.h
	source	create	 10 (5)	      aegis/aecd.c
	source	create	 2 (5)	      aegis/aecd.h
	source	create	 11 (5)	      aegis/aechown.c
	source	create	 2 (5)	      aegis/aechown.h
	source	create	 12 (5)	      aegis/aeclean.c
	source	create	 1 (5)	      aegis/aeclean.h
	source	create	 19 (5)	      aegis/aeclone.c
	source	create	 1 (5)	      aegis/aeclone.h
	source	create	 19 (5)	      aegis/aecp.c
	source	create	 2 (5)	      aegis/aecp.h

aecomplete/command/aegis.c
/*
 *	aegis - project change supervisor
 *	Copyright (C) 2002, 2003 Peter Miller;
 *	All rights reserved.
 *
 *	This program is free software; you can redistribute it and/or modify
 *	it under the terms of the GNU General Public License as published by
 *	the Free Software Foundation; either version 2 of the License, or
 *	(at your option) any later version.
 *
 *	This program is distributed in the hope that it will be useful,
 *	but WITHOUT ANY WARRANTY; without even the implied warranty of
 *	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *	GNU General Public License for more details.
 *
 *	You should have received a copy of the GNU General Public License
 *	along with this program; if not, write to the Free Software
 *	Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111, USA.
 *
 * MANIFEST: functions to manipulate aegiss
 */

#include <arglex2.h>
#include <command/aegis.h>
#include <command/private.h>
#include <complete/nil.h>
#include <complete/project/name.h>
#include <complete/project/name.h>


static complete_ty *result;


static void
destructor(command_ty *this_thing)
{
}


static void
redirect(const char *name)
{
    command_ty     *cmd;
    string_ty      *s;

    s = str_from_c(name);
    cmd = command_find(s);
    str_free(s);
    result = command_completion_get(cmd);
}


static void
aeb(void)
{
    redirect("aeb");
}


static void
aeca(void)
{
    redirect("aeca");
}


static void
aecd(void)
{
    redirect("aecd");
}


static void
aechown(void)
{
    redirect("aechown");
}


static void
aeclean(void)
{
    redirect("aeclean");
}


static void
aeclone(void)
{
    redirect("aeclone");
}


static void
aecp(void)
{
    redirect("aecp");
}


static void
aecpu(void)
{
    redirect("aecpu");
}


static void
aedb(void)
{
    redirect("aedb");
}


static void
aedbu(void)
{
    redirect("aedbu");
}


static void
aede(void)
{
    redirect("aede");
}


static void
aedeu(void)
{
    redirect("aedeu");
}


static void
aed(void)
{
    redirect("aed");
}


static void
aedn(void)
{
    redirect("aedn");
}


static void
aeib(void)
{
    redirect("aeib");
}


static void
aeibu(void)
{
    redirect("aeibu");
}


static void
aeifail(void)
{
    redirect("aeifail");
}


static void
aeipass(void)
{
    redirect("aeipass");
}


static void
aemv(void)
{
    redirect("aemv");
}


static void
aemvu(void)
{
    redirect("aemvu");
}


static void
aena(void)
{
    redirect("aena");
}


static void
aenbr(void)
{
    redirect("aenbr");
}


static void
aenbru(void)
{
    redirect("aenbru");
}


static void
aenc(void)
{
    redirect("aenc");
}


static void
aencu(void)
{
    redirect("aencu");
}


static void
aend(void)
{
    redirect("aend");
}


static void
aenf(void)
{
    redirect("aenf");
}


static void
aenfu(void)
{
    redirect("aenfu");
}


static void
aeni(void)
{
    redirect("aeni");
}


static void
aenpr(void)
{
    redirect("aenpr");
}


static void
aenrls(void)
{
    redirect("aenrls");
}


static void
aenrv(void)
{
    redirect("aenrv");
}


static void
aent(void)
{
    redirect("aent");
}


static void
aentu(void)
{
    redirect("aentu");
}


static void
aepa(void)
{
    redirect("aepa");
}


static void
aenpa(void)
{
    redirect("aenpa");
}


static void
aerpa(void)
{
    redirect("aerpa");
}


static void
aera(void)
{
    redirect("aera");
}


static void
aerd(void)
{
    redirect("aerd");
}


static void
aerm(void)
{
    redirect("aerm");
}


static void
aermu(void)
{
    redirect("aermu");
}


static void
aermpr(void)
{
    redirect("aermpr");
}


static void
aeri(void)
{
    redirect("aeri");
}


static void
aerrv(void)
{
    redirect("aerrv");
}


static void
aer(void)
{
    redirect("aer");
}


static void
aerb(void)
{
    redirect("aerb");
}


static void
aerbu(void)
{
    redirect("aerbu");
}


static void
aerfail(void)
{
    redirect("aerfail");
}


static void
aerpass(void)
{
    redirect("aerpass");
}


static void
aerpu(void)
{
    redirect("aerpu");
}


static void
aet(void)
{
    redirect("aet");
}


static void
aev(void)
{
    redirect("aev");
}


static void
ael(void)
{
    redirect("ael");
}


static void
usage(void)
{
    result = complete_nil();
}


static complete_ty *
completion_get(command_ty *cmd)
{
    static arglex_dispatch_ty dispatch[] =
    {
	{ arglex_token_build, aeb,},
	{ arglex_token_change_attributes, aeca,},
	{ arglex_token_change_directory, aecd,},
	{ arglex_token_change_owner, aechown,},
	{ arglex_token_clean, aeclean,},
	{ arglex_token_clone, aeclone,},
	{ arglex_token_copy_file, aecp,},
	{ arglex_token_copy_file_undo, aecpu,},
	{ arglex_token_develop_begin, aedb,},
	{ arglex_token_develop_begin_undo, aedbu,},
	{ arglex_token_develop_end, aede,},
	{ arglex_token_develop_end_undo, aedeu,},
	{ arglex_token_difference, aed,},
	{ arglex_token_delta_name, aedn,},
	{ arglex_token_integrate_begin, aeib,},
	{ arglex_token_integrate_begin_undo, aeibu,},
	{ arglex_token_integrate_fail, aeifail,},
	{ arglex_token_integrate_pass, aeipass,},
	{ arglex_token_move_file, aemv,},
	{ arglex_token_move_file_undo, aemvu,},
	{ arglex_token_new_administrator, aena,},
	{ arglex_token_new_branch, aenbr,},
	{ arglex_token_new_branch_undo, aenbru,},
	{ arglex_token_new_change, aenc,},
	{ arglex_token_new_change_undo, aencu,},
	{ arglex_token_new_developer, aend,},
	{ arglex_token_new_file, aenf,},
	{ arglex_token_new_file_undo, aenfu,},
	{ arglex_token_new_integrator, aeni,},
	{ arglex_token_new_project, aenpr,},
	{ arglex_token_new_release, aenrls,},
	{ arglex_token_new_reviewer, aenrv,},
	{ arglex_token_new_test, aent,},
	{ arglex_token_new_test_undo, aentu,},
	{ arglex_token_project_attributes, aepa,},
	{ arglex_token_project_alias_create, aenpa,},
	{ arglex_token_project_alias_remove, aerpa,},
	{ arglex_token_remove_administrator, aera,},
	{ arglex_token_remove_developer, aerd,},
	{ arglex_token_remove_file, aerm,},
	{ arglex_token_remove_file_undo, aermu,},
	{ arglex_token_remove_project, aermpr,},
	{ arglex_token_remove_integrator, aeri,},
	{ arglex_token_remove_reviewer, aerrv,},
	{ arglex_token_report, aer,},
	{ arglex_token_review_begin, aerb,},
	{ arglex_token_review_begin_undo, aerbu,},
	{ arglex_token_review_fail, aerfail,},
	{ arglex_token_review_pass, aerpass,},
	{ arglex_token_review_pass_undo, aerpu,},
	{ arglex_token_test, aet,},

	/*
	 * Then there are the more ambiguous arguments.
	 * These are only considered if none of the above are
	 * present on the command line (help least of all,
	 * because everything has help).
	 */
	{ arglex_token_version, aev, 1,},
	{ arglex_token_list, ael, 1,},
    };

    arglex2_retable(0);
    arglex_dispatch(dispatch, SIZEOF(dispatch), usage);
    return result;
}


static command_vtbl_ty vtbl =
{
    destructor,
    completion_get,
    sizeof(command_ty),
    "aegis",
};


command_ty     *
command_aegis()
{
    return command_new(&vtbl);
}
